src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -202,57 +200,76 @@ await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
+        // about the text changes between oldDocument and newDocument. By doing this, we can
+        // reduce the likelihood of the remote side encountering an unknown checksum and
+        // requiring a synchronization of the full document contents.
+        // This method uses JTF.RunAsync to create a fire-and-forget task. JTF.RunAsync will
+        // execute DispatchSynchronizeTextChangesHelperAsync synchronously until no longer
+        // possible. The hopefully common occurrence is that it is able to run synchronously
+        // all the way until it fires off the RPC call notifying the remote side of the changes.
+        _ = _threadingContext.JoinableTaskFactory.RunAsync(async () =>
         {
-            if (!oldDocument.TryGetText(out var oldText) ||
-                !newDocument.TryGetText(out var newText))
-            {
-                // we only support case where text already exist
-                continue;
-            }
-
-            // Avoid allocating text before seeing if we can bail out.
-            var changeRanges = newText.GetChangeRanges(oldText).AsImmutable();
-            if (changeRanges.Length == 0)
-                continue;
-
-            // no benefit here. pulling from remote host is more efficient
-            if (changeRanges is [{ Span.Length: var singleChangeLength }] && singleChangeLength == oldText.Length)
-                continue;
+            var wasSynchronized = await DispatchSynchronizeTextChangesHelperAsync().ConfigureAwait(false);
 
-            var state = await oldDocument.State.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
-            var newState = await newDocument.State.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
-
-            var textChanges = newText.GetTextChanges(oldText).AsImmutable();
-            builder.Add((oldDocument.Id, state.Text, textChanges, newState.Text));
-        }
+            var metricName = wasSynchronized ? "SucceededCount" : "FailedCount";
+            TelemetryLogging.LogAggregatedCounter(FunctionId.ChecksumUpdater_SynchronizeTextChangesStatus, KeyValueLogMessage.Create(m =>
+            {
+                m[TelemetryLogging.KeyName] = nameof(SolutionChecksumUpdater) + "." + metricName;
+                m[TelemetryLogging.KeyValue] = 1L;
+                m[TelemetryLogging.KeyMetricName] = metricName;
+            }));
 
-        if (builder.Count == 0)
             return;
 
-        await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
-            (service, cancellationToken) => service.SynchronizeTextChangesAsync(builder.ToImmutableAndClear(), cancellationToken),
-            cancellationToken).ConfigureAwait(false);
+            async Task<bool> DispatchSynchronizeTextChangesHelperAsync()
+            {
+                var client = await RemoteHostClient.TryGetClientAsync(_workspace, _shutdownToken).ConfigureAwait(false);
+                if (client == null)
+                    return false;
```

Comment:
slightly odd that this would show up as 'failure'.  this means a customer who has OOP off will show has massively failing this op.  i think this should maybe count as 'success' or perhaps NA so we don't get the wrong idea about being in a bad statee.